recipes/libdeflate/all/conanfile.py

```diff
@@ -60,51 +56,67 @@ def configure(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        if is_msvc(self) or self._is_clangcl:
+            vc = VCVars(self)
+            vc.generate()
+        else:
+            tc = AutotoolsToolchain(self)
+            tc.generate()
 
-    def _build_msvc(self):
-        with chdir(self, self._source_subfolder):
-            with vcvars(self), environment_append(VisualStudioBuildEnvironment(self).vars):
-                target = "libdeflate.dll" if self.options.shared else "libdeflatestatic.lib"
-                self.run("nmake /f Makefile.msc {}".format(target))
+    def _build_nmake(self):
+        with chdir(self, self.source_folder):
+            target = "libdeflate.dll" if self.options.shared else "libdeflatestatic.lib"
+            self.run(f"nmake /f Makefile.msc {target}")
 
     def _build_make(self):
-        autotools = AutoToolsBuildEnvironment(self, win_bash=(self._settings_build.os == "Windows"))
-        with chdir(self, self._source_subfolder):
+        autotools = Autotools(self)
+        with chdir(self, self.source_folder):
             autotools.make()
 
     def build(self):
         apply_conandata_patches(self)
         if is_msvc(self) or self._is_clangcl:
-            self._build_msvc()
+            self._build_nmake()
         else:
             self._build_make()
 
     def _package_windows(self):
-        self.copy("libdeflate.h", dst="include", src=self._source_subfolder)
+        copy(self, "libdeflate.h", dst=os.path.join(self.package_folder, "include"), src=self.source_folder)
         if self.options.shared:
-            self.copy("*deflate.lib", dst="lib", src=self._source_subfolder)
-            self.copy("*deflate.dll", dst="bin", src=self._source_subfolder)
+            copy(self, "*deflate.lib", dst=os.path.join(self.package_folder, "lib"), src=self.source_folder)
+            copy(self, "*deflate.dll", dst=os.path.join(self.package_folder, "bin"), src=self.source_folder)
         else:
-            self.copy("*deflatestatic.lib", dst="lib", src=self._source_subfolder)
+            copy(self, "*deflatestatic.lib", dst=os.path.join(self.package_folder, "lib"), src=self.source_folder)
 
     def _package_make(self):
-        autotools = AutoToolsBuildEnvironment(self, win_bash=(self._settings_build.os == "Windows"))
-        with chdir(self, self._source_subfolder):
-            autotools.install(args=["PREFIX={}".format(self.package_folder)])
+        autotools = Autotools(self)
+        with chdir(self, self.source_folder):
+            # TODO: replace by autotools.install() once https://github.com/conan-io/conan/issues/12153 fixed
+            # Note that PREFIX=/ also appears to be required on Linux.
+            autotools.install(args=["PREFIX=/", f"DESTDIR={unix_path(self, self.package_folder)}"])
```

Comment:
@jwillikers @uilianries continuing from other comment,

I get messages like:
```
install -m755 libdeflate.so.0 /build/conandata/conan-data/libdeflate/1.14/ccitest/test1/package/0582ee0920dad9fef631620d3ef80b963208d3a1/usr/local/lib;
```

Note it installs to /usr/local/lib

and then, `[HOOK - conan-center.py] post_package(): ERROR: [DEFAULT PACKAGE LAYOUT (KB-H013)] Unknown folder 'usr' in the package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H013)`

and ```CMake Error at build/generators/cmakedeps_macros.cmake:39 (message):
  Library 'deflate' not found in package.  If 'deflate' is a system library,
  declare it with 'cpp_info.system_libs' property
Call Stack (most recent call first):```

I know autotools adds PREFIX=/ to configure calls, but it didn't seem to be done this time.
Weirdly, it works on Windows without PREFIX=/

Reply:
Oh wait, this is a Makefile project and not actually Autotools, right? AutotoolsToolchain passes `--prefix` to the Autotools configure script. You wouldn't have that here so you'd have to use `PREFIX` or whatever the Makefile might expect. You can probably drop the `DESTDIR` then. I'm sorry for the confusion. We might need a dedicate generator for Makefile projects that doesn't require using things with `Autotools` in the name.

Reply:
Ok, dropped and just use PREFIX